src/efficient_ui.md

```diff
@@ -1,12 +1,12 @@
 # Efficient UI updates
 
-Relm4 follows the Elm programming model which means that data and widgets are separated. At first glance this might cause a problem. Larger applications need to efficiently update their widgets because rebuilding the whole UI for every update is not an option. But since data and widgets are separated, how do we know which UI elements need to be updated?
+Relm4 follows the Elm programming model which keeps the business logic and its data separated to the widgets. At the first glance you might think this causes problems, but larger applications need this decoupling as it enables them to efficiently modify their UI and business logic with keeping the needed changes local. Rebuilding the whole application with every change of the UI or logic is way too much effort. But since data and UI are separated, how do we know which UI elements need to be updated?
 
-Let's have a look at an imaginary example to visualize this problem. Imagine you have an app with 1000 counters and you only increment the first counter. The model receives the increment message for the first counter and increments it. Now the view function gets the updated model with 1000 counters and... well, has no idea what changed! So instead of one UI update we need to do 1000 because we don't know which of our counters was modified.
+Let's have a look at the following example: Imagine you have an app with 1000 counters and you only increment the first counter. The model receives the increment message for the first counter and increments it. Now the view function gets the updated model with 1000 counters and… well, has no idea what has changed! So instead of one UI update we need to do 1000 because we don't know which of our counters was modified.
 
-There are two concepts in Relm4 to avoid unnecessary UI updates
+There are two concepts in Relm4 to avoid unnecessary UI updates:
 
-+ Trackers: keep track of which struct fields were modified and only update the UI if they were modified.
-+ Factories: store data in a special data structures similar to the data structures in [`std::collections`](https://doc.rust-lang.org/std/collections/index.html) that will keep track of changes and will only apply minimal UI updates.
++ **Trackers** identify modifications of fields in `struct`s to only trigger the update the relevant UI elements.
++ **Factories** track changes in [`std::collections`](https://doc.rust-lang.org/std/collections/index.html) like data structures to perform also minimal UI updates. They are used to generate multiple similar widgets, e.g. a row of buttons, from a data collection.
```

Comment:
"`std::collections` like data structures" is not valid English AFAIK and should be replaced by "data structures similar to `std::collections`"